cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/docker/Validation.kt

```diff
@@ -0,0 +1,69 @@
+package com.cognifide.cogboard.storage.docker
+
+import com.cognifide.cogboard.CogboardConstants
+import com.cognifide.cogboard.storage.model.Board
+import com.fasterxml.jackson.databind.JsonMappingException
+import com.fasterxml.jackson.databind.MapperFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import io.vertx.core.json.JsonObject
+import java.util.stream.Collectors
+
+class Validation {
+    companion object {
+
+        private val mapper = jacksonObjectMapper().disable(MapperFeature.ALLOW_COERCION_OF_SCALARS)
+
+        fun validate(config: JsonObject): Boolean {
+            val boards = config.getJsonObject("boards")?.getJsonObject("boardsById")
+            return boards != null && validateBoards(boards)
+                    && hasWidgets(config)
+        }
+
+        private fun validateBoards(boardsData: JsonObject): Boolean {
+            val titles = mutableSetOf<String>()
+
+            val boards = mapJsonToObject(boardsData)
+
+            return boards?.all {
+                checkColumnsRange(it)
+                    && checkTitleLength(it)
+                    && checkTitleUnique(it, titles)
+                    && checkAutoSwitchInterval(it)
+            } ?: false
+        }
+
+        private fun mapJsonToObject(boards: JsonObject) =
+                try {
+                    boards.fieldNames().stream().map {
+                        val board = boards.getJsonObject(it)
+                        mapper.readValue<Board>(board.toString())
+                    }.collect(Collectors.toList())
+                } catch (e: JsonMappingException) {
```

Comment:
This way you swallow error message from mapper, which can be useful when debugging config. Consider letting out this exception :)

Reply:
Can you return some validation messages from this validator for the end user? You perform here a lot of validation!